repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2628,12 +2629,78 @@ private void addCategoriesToTermEntity(AttributeMutationContext ctx, List<Object
         RequestContext.get().endMetricRecord(metricRecorder);
     }
 
-    private void addMeaningsToEntity(AttributeMutationContext ctx, List<Object> createdElements, List<AtlasEdge> deletedElements) {
-        addMeaningsToEntity(ctx, createdElements, deletedElements, false);
+    /**
+     * Add meanings to entity relations.
+     * @param ctx
+     * @param createdElements
+     * @param deletedElements
+     *
+     * Notes : This case exists when user requests to add a meaning to multiple assets at a time.
+     *
+     */
+    private void addMeaningsToEntityRelations(AttributeMutationContext ctx, List<Object> createdElements, List<AtlasEdge> deletedElements) {
+        MetricRecorder metricRecorder = RequestContext.get().startMetricRecord("addMeaningsToEntityRelations");
+        List<AtlasVertex> assignedEntitiesVertices ;
+        List<String> currentMeaningsQNames ;
+        String newMeaningName = ctx.referringVertex.getProperty(NAME, String.class);
+        String newMeaningPropertyKey = ctx.referringVertex.getProperty(QUALIFIED_NAME, String.class);
+
+        // createdElements = all the entities user passes in assignedEntities
+        // it may contain entities which are already assigned to the relations
+        if (!createdElements.isEmpty()) {
```

Comment:
Please add TODO here for the Term.relationshipAttributes.assignedEntites case is not handled